scripts/geopmpy/update_report.py

```diff
@@ -0,0 +1,154 @@
+    global is_new_host
+    global is_end_section
+    result = []
+    # Keep track of global state
+    if old_line.startswith('Host: '):
+        is_new_host = True
+        is_end_section = False
+    if old_line.startswith('Region ') and is_new_host:
+        result.append('    Regions:')
+        is_new_host = False
+        is_end_section = False
+    if old_line.startswith('Epoch Totals:'):
+        is_new_host = False
+        is_end_section = True
+
+    # Define rules for rearranging report
+    if old_line.startswith('Host: '):
+        host_name = old_line.replace('Host: ', '')
+        result.append('  {}:'.format(host_name))
+    elif old_line == 'Region unmarked-region (0x00000000725e8066):':
+        result.append('    Unmarked Totals:')
+    elif old_line in ('Epoch Totals:', 'Application Totals:'):
+        result.append('    {}'.format(old_line))
+    elif old_line.startswith('Region '):
+        name_hash = old_line[len('Region '):-2]
+        sep = name_hash.rfind(' (')
+        name = name_hash[:sep]
+        hash = name_hash[sep + 2:]
+        result.append('    -')
+        result.append('      name: {}'.format(name))
+        result.append('      hash: {}'.format(hash))
+    elif old_line.startswith('    epoch-runtime-ignore (sec):'):
+        result.append('  {}'.format(old_line.replace('epoch-runtime-ignore', 'time-hint-ignore')))
+    elif old_line.startswith('    ignore-time (sec):'):
+        result.append('  {}'.format(old_line.replace('ignore-time', 'time-hint-ignore')))
+    elif old_line.startswith('    network-time (sec):'):
+        result.append(old_line.replace('network-time', '  time-hint-network'))
+        if not is_end_section:
+            result.append('      time-hint-ignore (s): 0')
+        result.extend(['      time-hint-compute (s): 0',
+                       '      time-hint-memory (s): 0',
+                       '      time-hint-io (s): 0',
+                       '      time-hint-serial (s): 0',
+                       '      time-hint-parallel (s): 0',
+                       '      time-hint-unknown (s): 0',
+                       '      time-hint-unset (s): 0'])
+    elif old_line.startswith('    geopmctl memory HWM:'):
+        num_byte = int(old_line.split(': ')[1].replace(' kB', '')) * 1024
+        result.append('      geopmctl memory HWM (B): {}'.format(num_byte))
+    elif old_line.startswith('    '):
+        result.append('  {}'.format(old_line))
+    elif not old_line.startswith(' '):
+        if is_new_host:
+            result.append('    {}'.format(old_line))
+        else:
+            result.append('  {}'.format(old_line))
+    result = [ll.rstrip().replace('0x00000000', '0x').replace('(sec)', '(s)').replace('(joules)', '(J)').replace('(watts)', '(W)') for ll in result]
+    return result
+
+def update_report_str(old_report):
+    if not old_report.startswith('##### geopm 1'):
+        raise ValueError('Input does not appear to be a legacy report')
```

Comment:
`1` -> `1.`   Will this script still be in use when GEOPM gets to version 10 ? :)